hibernate-core/src/main/java/org/hibernate/boot/model/naming/ImplicitNamingStrategyComponentPathImpl.java

```diff
@@ -30,14 +31,17 @@ protected String transformAttributePath(AttributePath attributePath) {
 	}
 
 	public static void process(AttributePath attributePath, StringBuilder sb) {
-		if ( attributePath.getParent() != null ) {
-			process( attributePath.getParent(), sb );
-			if ( StringHelper.isNotEmpty( attributePath.getParent().getProperty() ) ) {
-				sb.append( '_' );
-			}
-		}
-
 		String property = attributePath.getProperty();
+		final AttributePath parent = attributePath.getParent();
+		if ( parent != null && !parent.getProperty().equals( "" ) ) {
```

Comment:
I remember we have discussed the bad performance of `xxx.equals( "" )` and @Sanne has created a util method to avoid it. We can simply use `StringHelper.isNotEmpty( parent.getProperty )` here after parent's nullness checking.

Reply:
see https://github.com/hibernate/hibernate-orm/pull/3491

Reply:
Yes, that was burried in my comment at https://github.com/hibernate/hibernate-orm/pull/3583#issuecomment-705059462 but I completely agree it should use `#isNotEmpty`